enhancements/authentication/AuthConfig-missing-fields.md

```diff
@@ -0,0 +1,526 @@
+    // The exact value specified is used, so "/.well-known/openid-configuration"
+    // must be included in discoveryURL if needed.
+    // Example:
+    // discoveryURL: "https://oidc.oidc-namespace/.well-known/openid-configuration"
+    // discoveryURL: "https://oidc.example.com/.well-known/openid-configuration"
+    DiscoveryURL string `json:"discoveryURL,omitempty"`
+
+    // AudienceMatchPolicy controls how the "aud" claim in JWT tokens is validated.
+    // It allows flexible matching of the audience value in tokens.
+    // Possible values: MatchAny, MatchAll.
+    AudienceMatchPolicy AudienceMatchPolicyType `json:"audienceMatchPolicy,omitempty"`
+}
+```
+For more details on the Issuer type and its fields, see [here](https://github.com/openshift/api/blob/b8a067b12e1c404dc0f8e5dff9183ef20389318c/config/v1/types_authentication.go#L228-L252).
+
+#### TokenClaimMappings
+```go
+// TokenClaimMappings provides the claim mapping configuration for token-based identities
+type TokenClaimMappings struct {
+    // UID claim mapping
+    UID ClaimOrExpression `json:"uid,omitempty"`
+
+    // Extra claim mappings
+    Extra []ExtraMapping `json:"extra,omitempty"`
+}
+```
+For more details on the ClaimMappings type and its fields, see [here](https://github.com/openshift/api/blob/b8a067b12e1c404dc0f8e5dff9183ef20389318c/config/v1/types_authentication.go#L254-L265)
+
+
+#### TokenClaimValidationRule
+```go
+type TokenClaimValidationRule struct {
+    // Expression allows configuring a custom validation rule based on an expression
+    // This field defines a validation rule using a claim expression to evaluate the token
+    Expression string `json:"expression,omitempty"`
+
+    // Message defines a custom error message to be returned if the validation fails
+    Message string `json:"message,omitempty"`
+}
+```
+For more details on the ClaimValidationRule type and its fields, see [here](https://github.com/openshift/api/blob/b8a067b12e1c404dc0f8e5dff9183ef20389318c/config/v1/types_authentication.go#L440-L450)
+
+#### ToeknUserValidationRule
+```go
+// UserValidationRule provides the configuration for a single user validation rule.
+type ToeknUserValidationRule struct {
+	Expression string
+	Message    string
+}
+```
+For more details on the UserValidationRule type and its fields, see [here](https://github.com/kubernetes/kubernetes/blob/75909b89201386c8a555eadc79d14fb11f91747c/staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go#L284)
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Are there any unique considerations for making this change work with
+Hypershift?
+
+See https://github.com/openshift/enhancements/blob/e044f84e9b2bafa600e6c24e35d226463c2308a5/enhancements/multi-arch/heterogeneous-architecture-clusters.md?plain=1#L282
+
+How does it affect any of the components running in the
+management cluster? How does it affect any components running split
+between the management cluster and guest cluster?
```

Comment:
For these questions, I don't think there is any impact beyond what changes when the OIDC functionality that exists today is enabled.